src/test/java/org/amazon/testcases/TC002_GetItemsDetailsWithoutLogin.java

```diff
@@ -0,0 +1,98 @@
+	private final static String ELEMENTS_IN_ONE_PAGE = "//div[@class='a-section a-spacing-micro s-grid-status-badge-container']/a[1]";
+	private FileActions fileactions;
+	
+	@Test
+	public void searchItem() throws InterruptedException, IOException {
+		try {
+			fileactions = new FileActions();
+			home.selectCategory(PropertyReader.configReader("category"));
+			String[] items = PropertyReader.configReader("items").split(",");
+
+			for (String eachItem : items) {
+				getItem(eachItem);
+		    }
+		} 
+		catch (Exception e) {
+			e.printStackTrace();
+			System.out.println("------- Exception occurred -------" + e);
+		}
+	}
+
+	private void getItem(String eachItem) throws IOException {
+		
+		home.enterItemInTextbox(eachItem);
+		home.clickSubmit();
+		
+		WebDriverWait wait = new WebDriverWait(driver, 20);
+		wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(By.xpath(ELEMENTS_IN_ONE_PAGE)));
+		List<WebElement> itemsList = driver.findElements(By.xpath(ELEMENTS_IN_ONE_PAGE));
+		Set<String> mySet = new HashSet<String>();
+		Set<String[]> myFinalSet = new HashSet<String[]>();
+		
+		for (WebElement webElement : itemsList) {
+		    
+			if (mySet.size() >= 10) {
+				break;
+			}
+			else {
+				StringBuilder sb = new StringBuilder();
+				WebDriver newDriver = new ChromeDriver();
+				newDriver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+				String itemLink = webElement.getAttribute("href");
+				String hyperLink = "=HYPERLINK(\"" + itemLink + "\",\"GoToLink\")";
+				
+				if (itemLink.length() < 255) {
+					newDriver.get(itemLink);
+					
+					sb.append(eachItem).append(DELIMITER);
+					sb.append(newDriver.findElements(By.xpath(PRODUCT_TITLE)).isEmpty() ? "N/A" : newDriver.findElement(By.xpath(PRODUCT_TITLE)).getText()).append(DELIMITER);
+					sb.append(newDriver.findElements(By.xpath(BRAND)).isEmpty() ? "N/A" : newDriver.findElement(By.xpath(BRAND)).getText()).append(DELIMITER);
+					sb.append(newDriver.findElements(By.xpath(MRP)).isEmpty() ? "N/A" : newDriver.findElement(By.xpath(MRP)).getText().replace("\u20B9", "INR")).append(DELIMITER);
+					sb.append(newDriver.findElements(By.xpath(SALE_PRICE)).isEmpty() ? "N/A" : newDriver.findElement(By.xpath(SALE_PRICE)).getText().replace("\u20B9", "INR")).append(DELIMITER);
+					sb.append(newDriver.findElements(By.xpath(SAVINGS)).isEmpty() ? "N/A" : newDriver.findElement(By.xpath(SAVINGS)).getText().replace("\u20B9", "INR")).append(DELIMITER);
+					sb.append(newDriver.findElements(By.xpath(ITEM_WEIGHT)).isEmpty() ? "N/A" : newDriver.findElement(By.xpath(ITEM_WEIGHT)).getText()).append(DELIMITER);
+					sb.append(newDriver.findElements(By.xpath(NET_QUANTITY)).isEmpty() ? "N/A" : newDriver.findElement(By.xpath(NET_QUANTITY)).getText()).append(DELIMITER);
+					sb.append(newDriver.findElements(By.xpath(IN_STOCK_STATUS)).isEmpty() ? "No" : "Yes").append(DELIMITER);
+					sb.append(hyperLink);
+
+					mySet.add(sb.toString());
```

Comment:
Because there is nothing that will be duplicate, it does not make any sense to actually have a set here. This can be avoided. Rather than iterating the whole set again later in line no 92, you could add the data directly to final set.

Reply:
Done.